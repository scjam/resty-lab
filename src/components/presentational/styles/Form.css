.Form {
  margin-right: 0.3vw;
  width: 81.9vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.url {
  width: 100%;
  height: 25px;
  font-size: 12px;
  margin-top: 5px;
}
