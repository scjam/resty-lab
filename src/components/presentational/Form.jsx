/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles/Form.css';

const Form = ({ url, method, json, onSubmit, onChange }) => {
  return (
    <form onSubmit={onSubmit} className={styles.Form}>
      <input id="urlInput" type="text" name="url" placeholder="URL" value={url} onChange={onChange} />
      <section>
        <div>
          <label htmlFor="GET">
            <input id="GET" type="radio" name="method" value="GET" checked={method === 'GET'} onChange={onChange} />
            GET
          </label>
          <label>
            <input type="radio" name="method" value="PUT" checked={method === 'PUT'} onChange={onChange} />
            PUT
          </label>
          <label>
            <input type="radio" name="method" value="POST" checked={method === 'POST'} onChange={onChange} />
            POST
          </label>
          <label>
            <input type="radio" name="method" value="DELETE" checked={method === 'DELETE'} onChange={onChange} />
            DELETE
          </label>
        </div>
        <button>Ok!</button>
      </section>
      <textarea name="json" value={json} onChange={onChange}></textarea>
    </form>
  );
};

Form.propTypes = {
  url: PropTypes.string.isRequired,
  method: PropTypes.string.isRequired,
  json: PropTypes.string.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Form;
