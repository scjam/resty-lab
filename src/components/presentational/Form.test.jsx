import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Form from './Form';

describe('Form component', () => {
  afterEach(() => cleanup());
  it('renders Form', () => {
    const { asFragment } = render(<Form 
      url=""
      method=""
      json=""
      onSubmit={() => {}}
      onChange={jest.fn()} 
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
