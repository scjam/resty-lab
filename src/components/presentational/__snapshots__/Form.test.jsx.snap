// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<DocumentFragment>
  <main>
    <form>
      <input
        id="urlInput"
        name="url"
        placeholder="URL"
        type="text"
        value=""
      />
      <section>
        <div>
          <label
            for="GET"
          >
            <input
              id="GET"
              name="method"
              type="radio"
              value="GET"
            />
            GET
          </label>
          <label>
            <input
              name="method"
              type="radio"
              value="PUT"
            />
            PUT
          </label>
          <label>
            <input
              name="method"
              type="radio"
              value="POST"
            />
            POST
          </label>
          <label>
            <input
              name="method"
              type="radio"
              value="DELETE"
            />
            DELETE
          </label>
        </div>
        <button>
          Ok!
        </button>
      </section>
      <textarea
        name="json"
      />
    </form>
  </main>
</DocumentFragment>
`;
